spacy/pipeline/functions.py

```diff
@@ -136,3 +138,65 @@ def from_disk(self, path, **kwargs):
             "cfg": lambda p: self._set_config(srsly.read_json(p)),
         }
         util.from_disk(path, serializers, [])
+
+
+@Language.factory(
+    "doc_cleaner",
+    default_config={"attrs": {"tensor": None, "_.trf_data": None}, "silent": True},
+)
+def make_doc_cleaner(nlp: Language, name: str, *, attrs: Dict[str, Any], silent: bool):
+    return DocCleaner(attrs, silent=silent)
+
+
+class DocCleaner:
```

Comment:
Are we sure we don't want to inherit from `Pipe` here?

Reply:
Hmm, I just basically copied this from `token_splitter` above. I guess whatever we choose we should do for both?

Reply:
I guess there's an unclear line between what are helper functions/components or actual proper pipeline components. To me it feels like anything in `spacy.pipeline` should probably inherit from `Pipe`, but maybe not the type of functions that we have here. It doesn't really add much to inherit from `Pipe` either.

Reply:
There's something else that's drawn my attention though: currently `language.add_pipe` is typed to return  `Pipe`, ever since the mypy PR. Shouldn't this be `Callable[[Doc], Doc]`? Or should we have a protocol for this, just specifying the `__call__` signature? (maybe the latter is overkill)

Reply:
I also don't think it makes a difference to inherit from `Pipe`, although maybe it's slightly better to officially have `is_trainable`.

For components like these the main thing that would be nice is not to have to keep reimplementing the same kind of `cfg` serialization, but it's probably not enough of an issue to want to do anything in the core library about it.